lib/encoder.py

```diff
@@ -766,7 +770,11 @@ def _FilesToEncodings(self, files, videofile, bitrate=None,
         encoder or self._FileNameToEncoder(full_filename),
         bitrate or _FileNameToBitrate(full_filename),
         videofile or _FileNameToVideofile(full_filename))
-      candidate.Recover()
+      try:
+        candidate.Recover()
+      except Error:
```

Comment:
Re-iterating the more specific exception class catching: I glanced over the code and found that you directly throw Error() in many places. The style guide recommendation is to use a module's Error class only as base class for the actual exceptions thrown, so that you can catch either all exceptions from that module (Error) or any specific exception in a targeted manner.

Recommendation for a further refactoring, plus here add a TODO to make the catch more specific once you can.